src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/nonemptyatclausedescription/InputNonEmptyAtclauseDescriptionTwo.java

```diff
@@ -83,4 +83,22 @@ public int foo7() throws Exception
                 return 1;
         }
 
+    /**
+     *
+     * @param
+     *
+     */ // violation 2 lines above
```

Comment:
https://checkstyle.org/checks/javadoc/nonemptyatclausedescription.html#Violation_Messages

There are multiple violations, please specify violation message in comment, look around our code base there are bunch of examples 

Same for case below

Reply:
@romani just curious why we don't have all CIs in this PR

Reply:
> https://checkstyle.org/checks/javadoc/nonemptyatclausedescription.html#Violation_Messages
> 
> There are multiple violations, please specify violation message in comment, look around our code base there are bunch of examples
> 
> Same for case below

Done.

Another issue I think it should be considered, for `parameter` tag if there's a description without the **PARAMETER NAME** will not result a violation:
```
$ cat Test.java
public class Test {

  /**
   * @param some text
   */
  public int method(String str)
  {
     return 1;
  }
}

$ java -jar checkstyle-10.12.8-SNAPSHOT-all.jar -j Test.java

|--JAVADOC_TAG -> JAVADOC_TAG [3:5]
|   |--PARAM_LITERAL -> @param [3:5]
|   |--WS ->   [3:11]
|   |--PARAMETER_NAME -> some [3:12]  // HERE
|   |--WS ->   [3:16]
|   `--DESCRIPTION -> DESCRIPTION [3:17]
|       |--TEXT -> text [3:17]
|       |--NEWLINE -> \n [3:21]
|       |--LEADING_ASTERISK ->    * [4:0]
|       |--TEXT -> / [4:4]
|       |--NEWLINE -> \n [4:5]

```
I think the check should be updated so we extract the parameters from the AST and compare it with what we have after `param` tag.
I don't know if this deserve the effort, I'm just saying.

Reply:
I don't see messages on a line of `// violation`

Reply:
@romani After some investigation I found that we don't have a `checker` for **PARAMETER_NAME** or **CLASS_NAME** absence, additionally we don't have such messages in `messages.properties` (which means we never checked for this violation).
look here: 
[NonEmptyAtclauseDescriptionCheck](https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java)
we only check if the description exists or not, I think we need to extend the `check` to see if the `parameter` or `class` names are empty or not. 
if we are going to add such a check, it might be related to my previous suggestion https://github.com/checkstyle/checkstyle/pull/14803#discussion_r1569372055 to be considered too.

So, for now I think we can't add a `violation message` for this type of violation till we update the check.
This can't be done in this PR too, right? 
please tell me if missed anything.

Reply:
I meant to place violation message in same comment 
example: https://github.com/checkstyle/checkstyle/blob/bbabd4e3a93ba44e0fc89179e0b56472be28ad3b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodRecordsAndCompactCtors.java#L58

Reply:
> Another issue I think it should be considered, for parameter tag if there's a description without the PARAMETER NAME will not result a violation:

it is ok , there was no violation and will be no violation after your PR , and we need to create some Check eventually in separate issue and PR.

Reply:
> I meant to place violation message in same comment example:
> 
> https://github.com/checkstyle/checkstyle/blob/bbabd4e3a93ba44e0fc89179e0b56472be28ad3b/src/test/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocmethod/InputJavadocMethodRecordsAndCompactCtors.java#L58

This will require adding the new violation message in `messages.properties` and taking a reference from it to pass to [`testCheckTwo()`](https://github.com/checkstyle/checkstyle/blob/05981e0700b5aa7e7e84eb4f8a64b2f4cf88c648/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheckTest.java#L90) method, once I do this it conflicts here https://github.com/checkstyle/checkstyle/blob/05981e0700b5aa7e7e84eb4f8a64b2f4cf88c648/src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/NonEmptyAtclauseDescriptionCheck.java#L106

that's why I said we need first to add a check in `NonEmptyAtclauseDescriptionCheck.java` so i can send the new message.

Reply:
> If we have a violation already , we already have a message

remember that the absence of a parameter doesn't produce a violation, this PR solves a parsing error but still in the same check/violation **`empty description`**. so actually we don't have multiple violations here (:

Reply:
>There are multiple violations, please specify violation message

I did a typo here. It should be `There is violation, please specify violation message`

Reply:
@romani Done.
please tell me if you want me to open a new issue for the `parameter` absence as we discussed.

Reply:
@romani Ping